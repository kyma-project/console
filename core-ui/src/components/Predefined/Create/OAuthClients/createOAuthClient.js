export const createOAuthClient = (namespace, spec) => {
  const { name, grantTypes, responseTypes, secretName, scope } = spec;
  return {
    apiVersion: 'hydra.ory.sh/v1alpha1',
    kind: 'OAuth2Client',
    metadata: {
      name,
      namespace,
    },
    spec: {
      grantTypes,
      responseTypes,
      scope,
      secretName,
    },
  };
};
