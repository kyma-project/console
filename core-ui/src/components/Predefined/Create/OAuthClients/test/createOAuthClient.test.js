import { createOAuthClient } from '../createOAuthClient';

describe('createOAuthClient', () => {
  it('forms valid OAuth2 client', () => {
    const spec = {
      name: 'test-client',
      grantTypes: ['Grant', 'Michael', 'Jessica'],
      responseTypes: ['1'],
      secretName: 'test-secret-name',
      scope: 'a b',
    };
    expect(createOAuthClient('test-namespace', spec)).toMatchSnapshot();
  });
});
