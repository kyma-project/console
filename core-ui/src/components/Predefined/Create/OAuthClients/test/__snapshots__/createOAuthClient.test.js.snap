// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`createOAuthClient forms valid OAuth2 client 1`] = `
Object {
  "apiVersion": "hydra.ory.sh/v1alpha1",
  "kind": "OAuth2Client",
  "metadata": Object {
    "name": "test-client",
    "namespace": "test-namespace",
  },
  "spec": Object {
    "grantTypes": Array [
      "Grant",
      "Michael",
      "Jessica",
    ],
    "responseTypes": Array [
      "1",
    ],
    "scope": "a b",
    "secretName": "test-secret-name",
  },
}
`;
